src/main/java/org/traccar/geolocation/OpenCellIdGeolocationProvider.java

```diff
@@ -48,8 +48,14 @@ public void completed(JsonObject json) {
                         callback.onSuccess(
                                 json.getJsonNumber("lat").doubleValue(),
                                 json.getJsonNumber("lon").doubleValue(), 0);
+                    } else if (json.containsKey("error")) {
+                        String errorMessage = json.getJsonString("error").getString();
+                        if (json.containsKey("code")) {
+                            errorMessage += " Error code: " + json.getInt("code");
+                        }
+                        callback.onFailure(new GeolocationException(errorMessage));
                     } else {
-                        callback.onFailure(new GeolocationException("Coordinates are missing"));
+                        callback.onFailure(new GeolocationException("Coordinates are missing."));
                     }
```

Comment:
I would reorganize it into one else statement and then check "error" field inside and fallback to default if missing.